### ISSUE TITLE
Create Daohaus Connect Unconnected/Connected States

### ISSUE THREAD

**Issue by skuhlmann** [HausDAO/daohaus-monorepo#120]
# Overview

Build out the Daohaus Connect button and connection flow.

![Image](https://user-images.githubusercontent.com/6923345/166319062-fe090a0a-c7f3-4861-b49f-42489a76cbbe.png)

[figma link](https://www.figma.com/file/9lCwprCNaoAmUDmiMQpFEA/DAOhaus-v3?node-id=4005%3A28979)

# User stories

As a user I want to be able to connect my wallet to the daohaus apps.

As a developer I want to be able to add the DaohausConnect feature component to my application.

# Scenarios
```gherkin
Feature: Wallet Connect
Background: Given the user has set up a wallet.
Scenario: User has not connected their wallet to a supported chain.
    Given that I'm on the Hub App landing page
    And Wallet Connect button displays "Connect Wallet" as the Button text
    And the Profile Header is not displaying user profile data
    And the main content UI is not displaying the DAOs Table -- (need an empty state UI design)
    When I click on the "Connect Wallet" Button
    Then the UI displays data relevant to me
```

# Deliverables

- [ ] `<DaohausConnect />` component
- [ ] Use web3modal package for Wallet Connect
- [ ] Create a DaohausConnect context that can wrap a section of an app
- [ ] use @daohaus/dao-data to make a call to get user data (ceramic and ens)
- [ ] readme with code examples for context initialization and component usage

# Detail

**`<DaohausConnect />` component**
- Button component
- Avatar button component (in daohaus-ui?)
  - takes 'profile' data as a prop
  ```
  {
    address: '0x0',
    ens: 'dude.eth',
    image: 'not sure the format here yet',
    name: 'some user name',
  }
  ```
  - Fallbacks:
    - avatar: blockie
    - name: truncated address
    - address: (no ens: address) (ens: ens)
        - show ENS regardless if they have an ENS

- DropDownMenu component (in daohaus-ui?) when connected on correct network  
  - BlockExplorerLink component (add to daohaus-ui) - _**NEED TICKET**_
  -   - takes address
      - future proof for txhash as well - type (address or tx)
      - takes networkId
      - uses a keychain list in common-utilities to populate a link tag
- text indicate the connected network
- Disconnect button

**DaohausConnect context**
- Wrap web3modal
- expose some functions
  - connectWallet
  - disconnectWallet
  - getProfile
  - switchNetwork (this will take a target network and trigger metamask to switch to that network
- expose some data
  - address
  - connectedNetwork
  - provider
  - profile
  - isConnecting, isConnected (maybe useful?)
 

# Related Issues

[Button component](https://github.com/HausDAO/daohaus-monorepo/issues/96)
[ens data](https://github.com/HausDAO/daohaus-monorepo/issues/91)
[ceramic data](https://github.com/HausDAO/daohaus-monorepo/issues/92)

- [x] add a ticket for the wrong network use cases
- [x]  figure out where to put these states and delete this card:
https://github.com/orgs/HausDAO/projects/3/views/18?filterQuery=label%3A%22daohaus-connect%22

# Resources
- [blockie package](https://github.com/MyCryptoHQ/ethereum-blockies-base64)
- [web3modal](https://github.com/Web3Modal/web3modal)
- [pokemol](https://github.com/HausDAO/daohaus-app/blob/develop/src/contexts/InjectedProviderContext.jsx) injected provider context
- [pokemol switch network](https://github.com/HausDAO/daohaus-app/blob/develop/src/components/wrongNetworkToolTip.jsx)
- [quiver](https://github.com/raid-guild/quiver/tree/main/packages/quiver) has a similar implementation and in typescript

**Comment by skuhlmann** [comment]
Adding some detail on how you'll make the get profile call to the dao-data library. Should have this merged into `develop` by tomorrow.

To init the library you'll need a list of rpc endpoints for each supported network. These should also be required for setting up the context so you ought to have them handy.

example:
```typescript
import { Haus } from '@daohaus/dao-data'

  const rpcConfig = {
    '0x4': `https://${process.env['RIVET_KEY']}rinkeby.rpc.rivet.cloud`,
  };

 const haus = await Haus.create(rpcConfig);
```

then you'll have access to the `Profile` class with it's `get` method. The profile method takes the ceramic network name and a node url. we can use the test net for now. the get method takes an account address:

```typescript
profile = await haus.profile('testnet-clay', 'testnet-clay').get('0xEAC5F0d4A9a45E1f9FdD0e7e2882e9f60E301156)
```

**Comment by skuhlmann** [comment]
@jordanlesich 

Quick update on the @daohaus/data package - user profile function:

We're currently blocked on ceramic data fetch, but we've merged a PR that has the ens data so this should allow you to keep moving on this.

This spec should show you how you can init the package and fetch user data - slightly different than above and currently only returning ens
https://github.com/HausDAO/daohaus-monorepo/blob/develop/libs/dao-data/test/profile.spec.ts

```typescript
  const rpcConfig = {
    '0x4': `https://${process.env['NX_RIVET_KEY']}.rinkeby.rpc.rivet.cloud`,
    '0x64': 'https://rpc.gnosischain.com',
    '0x1': `https://${process.env['NX_RIVET_KEY']}.eth.rpc.rivet.cloud`,
  };
  
  haus = await Haus.create(rpcConfig);
  
 const profile = await haus.profile.get(
      '0x1223....'
);
```
returns
```
type AccountProfile = {
  address: string;
  ens: string | null;
};
```

**Comment by jordanlesich** [comment]
This will include an Avatar atom. 

I'm thinking about wrapping Quiver first, and if it doesn't work well, then I can go lower level.

**Comment by skuhlmann** [comment]
> I'm thinking about wrapping Quiver first, and if it doesn't work well, then I can go lower level.

Ok. If there are other dependencies in there that get annoying you could grab the quiver context and adjust if needed too.
